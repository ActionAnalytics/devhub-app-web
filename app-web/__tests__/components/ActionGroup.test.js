import React from 'react';
import { shallow } from 'enzyme';
import ActionGroup from '../../src/components/Cards/Backside/ActionGroup/ActionGroup';

describe('ActionGroup Component (card)', () => {
  it('matches snapshot', () => {
    const props = {
      title: 'Actions',
      actions: [
        {
          icon: <i>Icon</i>,
          actionName: 'foo',
          actionValue: 'bar',
        },
      ],
    };

    const wrapper = shallow(<ActionGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
