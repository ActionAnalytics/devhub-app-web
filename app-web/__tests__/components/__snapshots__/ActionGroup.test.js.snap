// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionGroup Component (card) matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionGroup
    actions={
      Array [
        Object {
          "actionName": "foo",
          "actionValue": "bar",
          "icon": <i>
            Icon
          </i>,
        },
      ]
    }
    title="Actions"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Actions
        </h3>,
        <ul>
          <Action
            actionName="foo"
            actionValue="bar"
            icon={
              <i>
                Icon
              </i>
            }
            link=""
          />
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Actions",
        },
        "ref": null,
        "rendered": "Actions",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Action
              actionName="foo"
              actionValue="bar"
              icon={
                <i>
                  Icon
                </i>
              }
              link=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "actionName": "foo",
              "actionValue": "bar",
              "icon": <i>
                Icon
              </i>,
              "link": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Actions
          </h3>,
          <ul>
            <Action
              actionName="foo"
              actionValue="bar"
              icon={
                <i>
                  Icon
                </i>
              }
              link=""
            />
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Actions",
          },
          "ref": null,
          "rendered": "Actions",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Action
                actionName="foo"
                actionValue="bar"
                icon={
                  <i>
                    Icon
                  </i>
                }
                link=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "actionName": "foo",
                "actionValue": "bar",
                "icon": <i>
                  Icon
                </i>,
                "link": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
